File: backend/scripts/telegramSession.js
import { TelegramClient } from "telegram";
import { StringSession } from "telegram/sessions/index.js";
import input from "input"; // npm i input
import 'dotenv/config';

const apiId = parseInt(process.env.TELEGRAM_API_ID);
const apiHash = process.env.TELEGRAM_API_HASH;
const stringSession = new StringSession(""); 

(async () => {
    console.log("Iniciando processo de login no Telegram...");
    
    if (!apiId || !apiHash) {
        console.error("TELEGRAM_API_ID ou TELEGRAM_API_HASH não encontrados no .env!");
        process.exit(1);
    }

    const client = new TelegramClient(stringSession, apiId, apiHash, {
        connectionRetries: 5,
    });
    
    await client.start({
        phoneNumber: async () => await input.text("Por favor, digite seu número de telefone (com DDI, ex: +5511999999999): "),
        password: async () => await input.text("Por favor, digite sua senha de verificação em duas etapas (se houver): "),
        phoneCode: async () => await input.text("Por favor, digite o código recebido no Telegram: "),
        onError: (err) => console.log(err),
    });
    
    console.log("Você está conectado agora.");
    const savedSession = client.session.save();
    console.log("\n==================================\n");
    console.log("Sua String de Sessão (Guarde-a no .env em TELEGRAM_SESSION):");
    console.log(savedSession);
    console.log("\n==================================\n");

    console.log("Carregando seus chats/canais recentes para descobrirmos o ID...");
    const dialogs = await client.getDialogs({});
    console.log("\nCanais/Grupos recentes:");
    for (const dialog of dialogs) {
        if (dialog.isChannel || dialog.isGroup) {
            console.log(`-> Nome: ${dialog.title} | ID: ${dialog.id}`);
        }
    }
    console.log("\nCopie o ID do canal desejado e cole no .env em TELEGRAM_CHANNEL_ID.");
    process.exit(0);
})();

File: backend/scripts/telegramSession.py
import os
from pathlib import Path
from pyrogram import Client
from dotenv import load_dotenv

env_path = Path(__file__).parent.parent.parent / '.env'
load_dotenv(env_path)

API_ID = os.getenv("TELEGRAM_API_ID")
API_HASH = os.getenv("TELEGRAM_API_HASH")

if not API_ID or not API_HASH:
    print("API_ID ou API_HASH não encontrados no .env!")
    exit(1)

script_dir = Path(__file__).parent.resolve()
session_name = str(script_dir / "my_account")

print("Iniciando Login no Telegram via Pyrogram...")
app = Client(session_name, api_id=int(API_ID), api_hash=API_HASH)

with app:
    print("\n✅ Sucesso! Você está logado no Pyrogram.")
    print("O arquivo de sessão my_account.session foi salvo.")
    print("Agora você pode rodar o robô de download sem se preocupar com códigos!")
